src/language/parser.ts

```diff
@@ -258,6 +271,16 @@ export class Parser {
    *   - InputObjectTypeDefinition
    */
   parseDefinition(): DefinitionNode {
+    const directives = this.parseDirectives(false);
+    // If a document-level SemanticNullability directive exists as
+    // the first element in a document, then all parsing will
+    // happen in SemanticNullability mode.
+    for (const directive of directives) {
+      if (directive.name.value === 'SemanticNullability') {
+        this._options.allowSemanticNullability = true;
+      }
+    }
```

Comment:
I think this would also be the first directive that applies to the parse phase right?

Reply:
Do you foresee that causing issues?

Reply:
I don't think it would no